pkg/manager/member/tiflash_upgrader.go

```diff
@@ -101,6 +111,25 @@ func (u *tiflashUpgrader) Upgrade(tc *v1alpha1.TidbCluster, oldSet *apps.Statefu
 			if store.State != v1alpha1.TiKVStateUp {
 				return controller.RequeueErrorf("tidbcluster: [%s/%s]'s upgraded TiFlash pod: [%s], store state is not UP", ns, tcName, podName)
 			}
+
+			needCheckStatus := false
+			tiflashVersion := tc.TiFlashVersion()
+			if _, ok := tiflashVersionsNeedCheckStatus[tiflashVersion]; ok {
+				needCheckStatus = true
+			} else if ver, err := semver.NewVersion(tiflashVersion); err == nil && ver.Compare(tiflashV512) >= 0 { // NOTE: if parse image version failed, will skip this check
+				needCheckStatus = true
+			}
+			if needCheckStatus {
+				status, err := u.deps.TiFlashControl.GetTiFlashPodClient(tc.Namespace, tc.Name, podName, tc.IsTLSClusterEnabled()).GetStoreStatus()
+				if err != nil {
+					return controller.RequeueErrorf("tidbcluster: [%s/%s]'s upgraded TiFlash pod: [%s], get store status failed: %s", ns, tcName, podName, err)
+				}
+
+				if status != tiflashapi.Running {
+					return controller.RequeueErrorf("tidbcluster: [%s/%s]'s upgraded TiFlash pod: [%s], store status is not Running", ns, tcName, podName)
```

Comment:
```suggestion
					return controller.RequeueErrorf("tidbcluster: [%s/%s]'s upgraded TiFlash pod: [%s], store status is %s instead of Running", ns, tcName, podName, status)
```